core/src/executor/evaluate/function.rs

```diff
@@ -574,6 +574,40 @@ pub fn sort<'a>(expr: Evaluated<'_>, order: Evaluated<'_>) -> Result<Evaluated<'
     }
 }
 
+pub fn slice<'a>(
+    name: String,
+    expr: Evaluated<'_>,
+    start: Evaluated<'_>,
+    length: Evaluated<'_>,
+) -> Result<Evaluated<'a>> {
+    let expr: Value = expr.try_into()?;
+    let mut start = eval_to_int!(name, start);
+    let length = match length.try_into()? {
+        Value::I64(number) => {
+            usize::try_from(number).map_err(|_| EvaluateError::FunctionRequiresUSizeValue(name))?
+        }
+        _ => {
+            return Err(EvaluateError::FunctionRequiresIntegerValue(name).into());
+        }
+    };
+    match expr {
+        Value::List(l) => {
+            if start < 0 {
+                start += l.len() as i64;
+            }
+            if start < 0 {
+                start = 0;
+            }
```

Comment:
really good! The slice function will soon become a new feature. It would be nice if only very small changes were possible. Couldn't these parts be put together?

Reply:
I intended that when a negative index is given, if index remain negative even after adding the size of vector to convert it to a positive index, it falls outside the range of vector. So I change index to 0.